go/stats/opentsdb/init.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opentsdb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+
+	"vitess.io/vitess/go/stats"
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/servenv"
+)
+
+var singletonBackend stats.PushBackend
+
+// Init attempts to create a singleton *opentsdb.backend and register it as a PushBackend.
+// If it fails to create one, this is a noop. The prefix argument is an optional string
+// to prepend to the name of every data point reported.
+func Init(prefix string) {
+	// Needs to happen in servenv.OnRun() instead of init because it requires flag parsing and logging
+	servenv.OnRun(func() {
+		log.Info("Initializing opentsdb backend...")
+		backend, err := InitWithoutServenv(prefix)
+		if err != nil {
+			log.Infof("Failed to initialize singleton opentsdb backend: %v", err)
+		} else {
+			singletonBackend = backend
+			log.Info("Initialized opentsdb backend.")
+		}
+	})
+	servenv.OnTerm(func() {
```

Comment:
This is new, and needed in order to flush gzip data when using file-based stats destination.